training/trainer.py

```diff
@@ -232,14 +232,17 @@ def train(
         tokenizer=tokenizer, mlm=False, return_tensors="pt", pad_to_multiple_of=8
     )
 
+    # enable fp16 if not bf16
+    fp16 = not bf16
+
     if not dbfs_output_dir:
         logger.warn("Will NOT save to DBFS")
 
     training_args = TrainingArguments(
         output_dir=local_output_dir,
         per_device_train_batch_size=per_device_train_batch_size,
         per_device_eval_batch_size=per_device_eval_batch_size,
-        fp16=False,
+        fp16=fp16,
```

Comment:
This probably won't matter now that it's set in the deepspeed config. Maybe doesn't hurt.

Reply:
This gives users an easy way to override the deepspeed config setting.